src/Cms/LanguageTranslations.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Exception;
+use Kirby\Data\Data;
+use Kirby\Filesystem\F;
+
+/**
+ * With helper methods provides get language translations or
+ * loads from custom `translations` root
+ * @since 5.0.0
+ *
+ * @package   Kirby Cms
+ * @author    Ahmet Bora <ahmet@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class LanguageTranslations
+{
+	protected App   $kirby;
+	protected array $data;
+
+	public function __construct(
+		protected Language $language,
+		self|array $translations = []
+	) {
+		$this->kirby = App::instance();
+		$this->setTranslations($translations);
+	}
+
+	/**
+	 * Returns a single translation string by key
+	 */
+	public function get(string $key, string $default = null): string|null
+	{
+		return $this->data[$key] ?? $default;
+	}
+
+	/**
+	 * Loads the language translations based on custom roots for provided language code
+	 */
+	public function load(array $default = []): array
+	{
+		if ($file = static::root()) {
+			try {
+				return Data::read($file);
+			} catch (Exception) {
+				return $default;
+			}
+		}
+
+		return $default;
+	}
+
+	/**
+	 * Saves the language translations in the custom root
+	 * @internal
+	 *
+	 * @return $this
+	 */
+	public function save(array $translations = []): static
+	{
+		if ($root = $this->root()) {
+			$this->data = $translations;
```

Comment:
I think this can call `$this->setTranslations($translations)` and be run even if there is no root for behavior consistency.